core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -651,16 +661,16 @@ public CompletableFuture<Optional<Throwable>> releaseAcquiredRecords(String memb
             for (Map.Entry<Long, InFlightBatch> entry : cachedState.entrySet()) {
                 InFlightBatch inFlightBatch = entry.getValue();
 
-                if (!inFlightBatch.memberId().equals(memberId)) {
-                    log.info("Member {} is not the owner of batch record {} for share partition: {}-{}",
-                            memberId, inFlightBatch, groupId, topicIdPartition);
-                    continue;
-                }
-
                 if (inFlightBatch.offsetState != null) {
                     log.trace("Offset tracked batch record found, batch: {} for the share partition: {}-{}", inFlightBatch,
                             groupId, topicIdPartition);
                     for (Map.Entry<Long, InFlightState> offsetState : inFlightBatch.offsetState.entrySet()) {
+                        // Check if member id is the owner of the offset.
+                        if (!offsetState.getValue().memberId.equals(memberId)) {
+                            log.info("Member {} is not the owner of offset: {} in batch: {} for the share"
```

Comment:
These log.info lines are probably helpful for debugging, but they cannot remain as info when we merge. Probably log.debug unless you want a normal user to be able to see them.

Reply:
makes sense, I'll use log.debug since mainly devs are going to use it in case things are not as expected.